ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/ApplicationPlugin.kt

```diff
@@ -86,11 +89,14 @@ public fun <P : Pipeline<*, ApplicationCall>, B : Any, F : Any> P.install(
     if (this is Route && plugin is RouteScopedPlugin) {
         return installIntoRoute(plugin, configure)
     }
+
+    check(this is Application)
```

Comment:
What other inheritants of `Pipeline<*, ApplicationCall>` do we have other than `Application` and `Route`?
Can we put 2 of them into a sealed class?
If not so -- why are we restricting plugin installation here? What if a user has some other Pipeline of ApplicationCall which is custom. Can they still install a plugin?

Reply:
Agree with @Ololoshechkin, we should not restrict it here. One example of such pipeline is `EnginePipeline` that has its own plugins

Reply:
Hey, @Ololoshechkin, @rsinukov. Thank you for the good notice! The reason to restrict it here is not to limit the receiver, but to get the application instance. I will try checking all possible cases here.